### PR TITLE
dev-lang/julia: adding new version 1.9.4

### PR DESCRIPTION
**Pull request by apraga**
- Adding patches for Julialang/julia#47862 (merged upstream but not yet available for 1.9.4)
- For compilation usr/lib64 is symlinked to usr/lib as julia expects library in lib64
- Installation fails to use string-replace on RELEASE_TARGET apparently due to string escaping issue in Makefile. This has been patched.

### CODE REVIEW

**Comment by xgqt** [comment]
@tamiko please review, see also: https://bugs.gentoo.org/show_bug.cgi?id=914841

File: dev-lang/julia/julia-1.9.4.ebuild [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+# Avoid building ocaml binding with LLVM_ENABLE_BINDINGS=OFF #47862 (merged upstream but not yet for 1.9.4)
+PATCHES=(
+	"${FILESDIR}"/julia-1.8.0-llvm-no-bindings.patch
+	"${FILESDIR}"/julia-1.9.4-string-replace.patch
+	)
+QA_FLAGS_IGNORED+='usr/.*/julia/sys.so'  # Julia sysimage generated by bootstrapping.
+
+# Huge thanks to Arch Linux developers for the patches.
+archlinux_uri="https://raw.githubusercontent.com/archlinux/svntogit-community/packages/julia/trunk/"
+archlinux_patches=(
```
Comment (xgqt) [inline dev-lang/julia/julia-1.9.4.ebuild:73]: so this code is dead now, plz remove

File: dev-lang/julia/julia-1.9.4.ebuild [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+	sed -i \
+		-e "\|SHIPFLAGS :=|c\\SHIPFLAGS := ${CFLAGS}" \
+		Make.inc || die
+
+	sed -i \
+		-e "s|ar -rcs|$(tc-getAR) -rcs|g" \
+		src/Makefile || die
+
+	# disable doc install starting	git fetching
+	sed -i -e 's~install: $(build_depsbindir)/stringreplace $(BUILDROOT)/doc/_build/html/en/index.html~install: $(build_depsbindir)/stringreplace~' Makefile || die
```
Comment (xgqt) [inline dev-lang/julia/julia-1.9.4.ebuild:122]: I think we can put this in a patch, can we? or wqill it get overwritten by Julia's makefile magic?

Comment (apraga) [reply dev-lang/julia/julia-1.9.4.ebuild]: A patch has been added instead.

**Comment by xgqt** [comment]
@apraga Could you list any Julia pkgs you tried to use this release with? I think Geany, Gtk and DiscreteMarkovChains were good test targets, mainly to see if Julia correctly finds all required libs.

**Comment by apraga** [comment]
@xgqt I've tested it with 
- CairoMakie,GLMakie,AlgebraOfGrahpics (my code)
- CSV, DataFramesMeta (my code)
- GTK (the tutorial)
- DiscreteMarkovChains (the tutorial)

I could not find a Geany package on Google.

By the way, I'm having trouble generating a correct manifest as it sometimes include .ebuild and patch files. The current version passes the QI.

**Comment by xgqt** [comment]
> @xgqt I've tested it with
> 
>     * CairoMakie,GLMakie,AlgebraOfGrahpics (my code)
> 
>     * CSV, DataFramesMeta (my code)
> 
>     * GTK (the tutorial)
> 
>     * DiscreteMarkovChains (the tutorial)
> 

That's a lot of work, thanks a lot!

> I could not find a Geany package on Google.

Hmm, weird, maybe it was renamed.

> By the way, I'm having trouble generating a correct manifest as it sometimes include .ebuild and patch files. The current version passes the QI.

I think you are running `ebuild manifest` on a non-official git repo... which is a bit weird

**Comment by xgqt** [comment]
If you squash the commits into one I can merge this.

**Comment by apraga** [comment]
> > By the way, I'm having trouble generating a correct manifest as it sometimes include .ebuild and patch files. The current version passes the QI.
> 
> I think you are running `ebuild manifest` on a non-official git repo... which is a bit weird
Running `ebuild manifest` in /var/db/repos/gentoo/dev-lang/julia added these extra files in the Manifest. Maybe that's not the proper way to do it :) 

Commits are squashed, thanks for guiding me in this !

**Pull request closed without merging**